hydra-node/src/Hydra/Network/Reliability.hs

```diff
@@ -136,67 +180,68 @@ withReliability ::
   (MonadThrow (STM m), MonadThrow m, MonadAsync m) =>
   -- | Tracer for logging messages.
   Tracer m ReliabilityLog ->
+  -- | Our persistence handle
+  MessagePersistence m msg ->
   -- | Our own party identifier.
   Party ->
-  -- | Other parties' identifiers.
-  [Party] ->
+  -- | All parties' identifiers.
+  Vector Party ->
```

Comment:
We should revert back to using `me` + `others` and sort ourselves rather then requiring the caller to be able to construct that `Vector`